### DISCUSSION TITLE
gitcookie

### DISCUSSION THREAD

**Discussion by pburgisser** [argoproj/argo-cd#11890] [Q&A]
Hi folks,

My primary GIT server authenticates users with a gitcookie and it's the only way to authenticate.
I looked through the doc but I don't find how I could connect my ArgoCD to my repositories.

I also run a secondary GIT server based on Gitlab but it's behind a GCP Identity Aware Proxy, I could easily create a service account for ArgoCD and use some sort of authentication headers (bearer token) but again, I don't see how to do that with ArgoCD.

Does anyone had the same challenges ?

Thanks a lot

**Comment by jannfis** [comment]
I never used gitcookies, but from what I've read, you may have success if you

* Mount your gitcookies file to the `argocd-repo-server` pod and
* Run `git config --global ...` to set your gitcookies file globally from an init container on the `argocd-repo-server` pod

Then configure your repository without any authentication whatsoever. I'm not sure if this would work, but it could be worth a try.

**Comment by pburgisser** [comment]
Apparently, when you do git config --global goes into /home/argocd/.gitconfig which is ignored by go-git library.
I tried to put the config in /etc/gitconfig (equivalent to git config --system) but looks like go-git doesn't know what to do with that. 

However, from the container' shell, `git clone` works. I believe it's in the go-git library :(

Cheers for the tip!

**Reply by jannfis** [reply]
OK, so after researching a little, `gitcookies` seems to be really just a plain text file that stores HTTP cookie data to be transmitted in the requests that the Git client makes to the server.

I think it should be possible to have both, go-git and the fork/exec Git client consume the same file. This would obviously require some non-trivial code changes, but feel free to file an enhancement request to support gitcookies authentication.

**Comment by ssmall** [comment]
I'm encountering the same issue as @pburgisser described in the original post, and wondering if there was ever an enhancement request filed to support this functionality. I wasn't able to find one by searching existing issues.
